### PR TITLE
Webhooks Backend (based on #90)

### LINKED ISSUE #88: Webhook Backend is not started
## Current Behavior

Webhooks are configured but don't fire.

...

## Expected Behavior

Webhooks should fire.

...

## Debug Information

I'm running `v2.4.2-ldap` and the rqworker (webhook backend) is not being started.

See https://github.com/digitalocean/netbox/issues/2338

### PR DESCRIPTION
**Pull request by cimnine**
First: This is based on the great work by @bdlamprecht in #90!

This PR would resolve #88.

Why did I open another PR for the same feature?
**tl;dr:** I believe to have optimized Brady's work, but it's a different approach.

Brady's approach was running the worker (the process that actually sends out the webhooks, i.e. it connects to the configured server via http) in the same container as the main netbox process.

Now the main thing I did in this PR was to extract the worker into it's own container. I see almost only benefits with my new approach:

* Logs of Netbox and it's webhooks worker are separate.
* In case one needs to scale either of the containers, they can be scaled independently.
* The worker has a much faster startup time, as it doesn't need to wait for Netbox DB to be initialized.
* `docker-compose ps` will tell you right away if the worker is running (vs. checking supervisord somehow).

What is still the same
* Netbox and the worker still run from the same image
* `docker-compose up` still automatically starts Netbox _and_ the worker.

Other changes compared to #90:
* No redis password (as I don't see the benefit of putting one)
* Webhooks are disabled for existing setups by default. (This is important because one will have to touch docker-compose.yml in order to get this feature even working.)
  * For new setups, webhooks are enabled by default via `netbox.env`
* Some cleaning up in nginx config, Dockerfile, etc. See the individual commit messages.

### CODE REVIEW

**Comment by bdlamprecht** [comment]
I like the idea of having a different container for running the `netbox-worker` so you can see if it is running or not. I didn't even think about starting a new `netbox` container but not populate the DB, pretty neat approach.

The only concern I have is this statement:
>No redis password (as I don't see the benefit of putting one)

I have a feeling a lot of other people are going to disagree with this.
Yes, I know the data isn't the same type of importance as what is in `postgres`, however, it _could_ be used nefariously, so I'd prefer to keep the password there if at all possible just to be careful.

Other than that, looks good and awaiting the merge into `master`.

**Comment by cimnine** [comment]
About the redis password: It's not that you _can't_ configure a redis password.

But in the default setup, Redis is not exposed to the outside world anyway. So it's the same if there's no password or a static password. Isn't it?

PG has a password, because I think it's not possible to run the postgres container without password.

**Comment by cimnine** [comment]
On second thought, I'll add a default password. It's not safer than no password, but it's an example on how to set a password.

**Comment by bdlamprecht** [comment]
Yes, I think showing an example of how a password for `redis` is setup would be beneficial.
It caused me a lot of time and great angst, as I stated before, to get it to work.

Ultimately, I had to fall back to a workaround that the developer of the official `redis` on DockerHub suggested.  I can only imagine many other people would struggle to do the same thing.

Also, just a quick note, I just tried to clone your updated `webhooks-backend` branch and the `nginx` container died.  I didn't look into why, but thought I would mention it.

**Comment by cimnine** [comment]
> Also, just a quick note, I just tried to clone your updated webhooks-backend branch and the nginx container died. I didn't look into why, but thought I would mention it.

I think you need to remove the `netbox-nginx-config` volume, as the config in it was updated.

**Comment by cimnine** [comment]
@bdlamprecht did this resolve?

> Also, just a quick note, I just tried to clone your updated webhooks-backend branch and the nginx container died. I didn't look into why, but thought I would mention it.

**Comment by cimnine** [comment]
> @bdlamprecht did this resolve?

Confirmed via PN in [networktocode slack](networktocode.slack.com).

File: Dockerfile [hunk 1]
```diff
@@ -21,7 +21,13 @@ RUN pip install \
 # napalm is used for gathering information from network devices
       napalm \
 # ruamel is used in startup_scripts
-      ruamel.yaml
+      ruamel.yaml \
+# pinning django to the version required by netbox
+# adding it here, to install the correct version of
+# django-rq
+      'Django>=1.11,<2.1' \
```
Comment (neckhair) [inline Dockerfile]: You could make this dynamic with a default.

Comment (cimnine) [reply Dockerfile]: I don't get what you mean, could you elaborate?

Comment (neckhair) [reply Dockerfile]: Sorry, my eyes were already too tired.. They just saw the one version number 🤦‍♂️
